cylc/flow/task_events_mgr.py

```diff
@@ -778,6 +778,7 @@ def _process_message_check(
 
         if (
                 itask.state(TASK_STATUS_WAITING)
+                and itask.tdef.run_mode == 'live'
```

Comment:
Disables logging?

Reply:
The code within this section is designed to avoid problems with polling - but there will be no polling in simulation mode, and it causes simulated tasks to hang in waiting mode.

Reply:
But this is the process message check run on incomming messages, it doesn't trigger polling?!

Reply:
Comment at line 799 

>             # Ignore messages if task has a retry lined up
            # (caused by polling overlapping with task failure)
            
It's been accepted on master, but I can't find good enough notes on why, will work it out again.

Reply:
I've pushed up this change to see if I fixed this issue.